code/modules/clothing/modular_armor/attachments/modules.dm

```diff
@@ -475,6 +475,93 @@
+	attach_features_flags = ATTACH_REMOVABLE|ATTACH_ACTIVATION|ATTACH_APPLY_ON_MOB
+	slot = ATTACHMENT_SLOT_MODULE
+	toggle_signal = COMSIG_KB_ARMORMODULE
+	COOLDOWN_DECLARE(mirage_cooldown)
+
+/obj/item/armor_module/module/mirage/activate(mob/living/user)
+	if(!COOLDOWN_CHECK(src, mirage_cooldown))
+		balloon_alert(user, "[COOLDOWN_TIMELEFT(src, mirage_cooldown)*0.1] seconds")
+		return
+	var/alpha_mod = user.alpha * 0.8
+	user.alpha -= alpha_mod
+	var/mob/illusion/mirage_nade/fake = new(get_turf(user), user, null, 15 SECONDS)
+	addtimer(CALLBACK(src, PROC_REF(end_mirage), user, alpha_mod, fake), 15)
+	COOLDOWN_START(src, mirage_cooldown, 30 SECONDS)
+
+/// just cleans up the alpha on both the user and the fake
+/obj/item/armor_module/module/mirage/proc/end_mirage(mob/user, alpha_mod, mob/illusion/mirage_nade/fake)
+	user.alpha += alpha_mod
+	fake.alpha = user.alpha
+
+
+#define ARMORLOCK_DURATION 6 SECONDS
+#define ARMORLOCK_SIEMENS_COEFF -0.9
+#define ARMORLOCK_PERMEABILITY_COEFF -1
+#define ARMORLOCK_GAS_TRANSFER_COEFF -1
+
+/obj/item/armor_module/module/armorlock
+	name = "\improper Thor Armorlock Module"
+	desc = "Designed for mounting on modular armor. This module seals gaps in the armor when activated, making the user unable to do any actions but increasing their armor."
+	icon = 'icons/mob/modular/modular_armor_modules.dmi'
+	icon_state = "mod_armorlock"
+	worn_icon_state = "mod_armorlock_a"
+	slowdown = 0.1
+	attach_features_flags = ATTACH_REMOVABLE|ATTACH_ACTIVATION|ATTACH_APPLY_ON_MOB
+	slot = ATTACHMENT_SLOT_MODULE
+	toggle_signal = COMSIG_KB_ARMORMODULE
+	COOLDOWN_DECLARE(armorlock_cooldown)
+	///This is the armor amounts we will be adding and removing when armor lock is activated
+	var/datum/armor/locked_armor_mod = list(MELEE = 50, BULLET = 50, LASER = 50, ENERGY = 50, BOMB = 50, BIO = 50, FIRE = 50, ACID = 50)
+
+/obj/item/armor_module/module/armorlock/Initialize(mapload)
+	. = ..()
+	locked_armor_mod = getArmor(arglist(locked_armor_mod))
+
+/obj/item/armor_module/module/armorlock/Destroy()
+	. = ..()
+	locked_armor_mod = null
+
+/obj/item/armor_module/module/armorlock/activate(mob/living/user)
+	if(!COOLDOWN_CHECK(src, armorlock_cooldown))
+		balloon_alert(user, "[COOLDOWN_TIMELEFT(src, armorlock_cooldown)*0.1] seconds")
+		return
+
+	user.add_traits(list(TRAIT_HANDS_BLOCKED, TRAIT_STOPS_TANK_COLLISION, TRAIT_IMMOBILE, TRAIT_INCAPACITATED), REF(src))
+	user.move_resist = MOVE_FORCE_OVERPOWERING
+	user.log_message("has been armor locked for [ARMORLOCK_DURATION] ticks", LOG_ATTACK, color="pink")
+
+	var/image/shield_overlay = image('icons/effects/effects.dmi', null, "shield-blue")
+	user.overlays += shield_overlay
+	parent?.soft_armor = parent?.soft_armor.attachArmor(locked_armor_mod)
+	playsound(user, 'sound/items/armorlock.ogg', 50)
+
+	parent.siemens_coefficient += ARMORLOCK_SIEMENS_COEFF
+	parent.permeability_coefficient += ARMORLOCK_PERMEABILITY_COEFF
+	parent.gas_transfer_coefficient += ARMORLOCK_GAS_TRANSFER_COEFF
```

Comment:
None of this actually helps the user. Armour valies apply on equip/unequip, so the user is uneffected. These should be applied to the mob directly.